datastore/elastic_search/query.py

```diff
@@ -318,3 +435,82 @@ def _parse_es_ngram_search_results(ngram_results, ngrams_length):
 
     return variant_dictionary
+
+
+def _parse_es_search_results(results):
+    """
+    Parses highlighted results returned from elasticsearch query on user typed sentence
+
+    Args:
+        results: search results dictionary from elasticsearch including highlights and scores
+
+    Returns:
+        dictionary of the parsed results from highlighted search query results on ngrams_list
+
+    Example:
+        Parameter ngram_results has highlighted search results as follows:
+
+        {u'_shards': {u'failed': 0, u'successful': 5, u'total': 5},
+        u'hits': {u'hits': [{u'_id': u'AVrW02UE9WNuMIY9vmWn',
+        u'_index': u'doc_type_name',
+        u'_score': 11.501145,
+        u'_source': {u'dict_type': u'variants',
+        u'entity_data': u'city',
+        u'value': u'goa',
+        u'variants': [u'', u'goa']},
+        u'_type': u'data_dictionary',
+        u'highlight': {u'variants': [u'<em>goa</em>']}},
+        {u'_id': u'AVrW02W99WNuMIY9vmcf',
+        u'_index': u'gogo_entity_data',
+        u'_score': 11.210829,
+        u'_source': {u'dict_type': u'variants',
+        u'entity_data': u'city',
+        u'value': u'Mumbai',
+        u'variants': [u'', u'Mumbai']},
+        u'_type': u'data_dictionary',
+        u'highlight': {u'variants': [u'<em>Mumbai</em>']}},
+        ...
+        u'max_score': 11.501145,
+        u'total': 17},
+        u'timed_out': False,
+        u'took': 96}
+
+        After parsing highlighted results, this function returns
+
+        {...
+         u'Mumbai': u'Mumbai',
+         ...
+         u'goa': u'goa',
+         u'mumbai': u'mumbai',
+         ...
+        }
+
+    """
+    entity_value_list, variant_value_list = [], []
+    variant_dictionary = {}
+    if results and results['hits']['total'] > 0:
+        for hit in results['hits']['hits']:
+            if 'highlight' not in hit:
+                continue
+            count_of_variants = len(hit['highlight']['variants'])
+            count = 0
+            while count < count_of_variants:
+                entity_value_list.append(hit['_source']['value'])
+                variant_value_list.append(hit['highlight']['variants'][count])
+                count += 1
+    count = 0
+
+    while count < len(variant_value_list):
+        variant_value_list[count] = re.sub('\s+', ' ', variant_value_list[count]).strip()
+        if variant_value_list[count].count('<em>') == len(variant_value_list[count].split()):
+            variant = variant_value_list[count].replace('<em>', '')
+            variant = variant.replace('</em>', '')
+            if variant.strip() in variant_dictionary:
+                existing_value = variant_dictionary[variant.strip()]
+                if len(existing_value.split(' ')) > len(entity_value_list[count].split(' ')):
+                    variant_dictionary[variant.strip()] = entity_value_list[count]
+            else:
+                variant_dictionary[variant.strip()] = entity_value_list[count]
+        count += 1
```

Comment:
Optional, The body of the method can be refactored to be more Pythonic by using `enumerate()`